base/parse.jl

```diff
@@ -194,3 +194,5 @@ function parse(str::AbstractString; raise::Bool=true)
     end
     return ex
 end
+
+promote_op{T}(::typeof(parse), ::Type{T}, ::Any) = T
```

Comment:
This should be removed. Since the first argument to `parse` is a type, it would have to be `Type{Type{T}}`, but that won't be useful very often since an array of types would have element type `DataType`.